src/app/routers/schema_router.py

```diff
@@ -0,0 +1,114 @@
+import exception.exceptions as exceptions
+from fastapi import APIRouter, Body, Depends
+from logging_config import logging
+from models.schema_models import Schema, SchemaMetadata, SchemaMetadataWithGuid
+from repositories.buckets.schema_bucket_repository import SchemaBucketRepository
+from repositories.firebase.schema_firebase_repository import SchemaFirebaseRepository
+from services.schema.schema_processor_service import SchemaProcessorService
+from services.validators.query_parameter_validator_service import (
+    QueryParameterValidatorService,
+)
+
+router = APIRouter()
+
+logger = logging.getLogger(__name__)
+
+
+@router.post("/v1/schema", response_model=SchemaMetadataWithGuid)
+async def post_schema_metadata(
+    schema_metadata: Schema = Body(...),
+    schema_processor_service: SchemaProcessorService = Depends(),
+) -> SchemaMetadataWithGuid:
+    """
+    Posts the schema metadata to be processed.
+
+    Parameters:
+    schema_metadata (Schema): schema metadata to be processed.
+    schema_processor_service (SchemaProcessorService): injected processor service for processing the schema metadata.
+    """
+    logger.info("Posting schema metadata...")
+    logger.debug(f"Input body: {{{schema_metadata}}}")
+
+    posted_schema_metadata = schema_processor_service.process_schema_metadata(
+        schema_metadata
+    )
+
+    logger.info("Schema metadata successfully posted.")
+    logger.debug(f"Schema metadata: {posted_schema_metadata}")
+
+    return posted_schema_metadata
+
+
+@router.get("/v1/schema")
+async def get_schema_metadata_from_bucket(
```

Comment:
I might be wrong here, but I believe that this endpoint is to retrieve the whole schema of a survey from JSON file instead of only the metadata, which the metadata only contains 5 attributes while response here contains the metadata and the actual schema in attribute properties as well. So I am curious about the changes in the return annotation and naming